### PR TITLE
feat: search query builder

### PR DESCRIPTION
**Pull request by LeoniePeters**

### CODE REVIEW

**Review by lbiedinger — APPROVED** [review]

File: packages/portal/src/components/search/SideFilters.vue [hunk 1]
```diff
@@ -13,6 +13,16 @@
     >
       <section role="search">
         <client-only>
+          <b-row
+            v-if="advancedSearchEnabled"
+          >
+            <b-button
+              @click="toggleAdvancedSearch"
+            >
+              {{ $t('search.advanced.show', {'show': showAdvancedSearch ? 'hide' : 'show' }) }}
```
Comment (rwd) [inline packages/portal/src/components/search/SideFilters.vue:22]: Add a space before first `'show'`

File: packages/portal/src/components/search/SideFilters.vue [hunk 2]
```diff
@@ -511,6 +524,9 @@
       toggleFilterSheet() {
         this.$store.commit('search/setShowFiltersSheet', !this.$store.state.search.showFiltersSheet);
       },
+      toggleAdvancedSearch() {
+        this.$store.commit('search/setShowAdvancedSearch', !this.$store.state.search.showAdvancedSearch);
```
Comment (rwd) [inline packages/portal/src/components/search/SideFilters.vue:528]: Let's not use the store for this, but instead emit an event for SearchInterface to react to.

Alternatively, even that may not be needed if the toggle button is moved into the slot.

File: packages/portal/src/components/search/SearchQueryBuilder.vue [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    name: 'SearchQueryBuilder',
+
+    components: {
+      SearchQueryBuilderRule
+    },
+
+    data() {
+      return {
+        queryRules: [{}],
+        searchFields: [
```
Comment (rwd) [inline packages/portal/src/components/search/SearchQueryBuilder.vue:44]: If these are always the same, does it make sense for them to be here, and a prop on SearchQueryBuilderRule, or should they just be static local data in that component?

**Review by rwd — CHANGES_REQUESTED** [review]

File: packages/portal/src/components/search/SearchQueryBuilder.vue (hunk 3 above)
Comment (LeoniePeters) [reply packages/portal/src/components/search/SearchQueryBuilder.vue:44]: Will be done on other branch: https://github.com/europeana/portal.js/tree/feat/EC-6313-searchQueryBuilder-parsing

File: packages/portal/src/components/search/SearchQueryBuilder.vue [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+          // TODO: replace values with query syntax
+          { value: null, text: this.$i18n.t('search.advanced.selectOne') },
+          { value: 'a', text: this.$t('search.advanced.fields.anyField') },
+          { value: 'b', text: this.$t('search.advanced.fields.subject') },
+          { value: 'c', text: this.$t('search.advanced.fields.title') },
+          { value: 'd', text: this.$t('search.advanced.fields.description') },
+          { value: 'e', text: this.$t('search.advanced.fields.creator') },
+          { value: 'f', text: this.$t('search.advanced.fields.type') }
+        ],
+        modifiers: [
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/components/search/SearchQueryBuilder.vue:54]: Will be done on other branch: https://github.com/europeana/portal.js/tree/feat/EC-6313-searchQueryBuilder-parsing

File: packages/portal/src/components/search/SideFilters.vue (hunk 2 above)
Comment (LeoniePeters) [reply packages/portal/src/components/search/SideFilters.vue]: Toggle button moved into the slot and state handled in `SearchInterface`.

File: packages/portal/src/components/search/SideFilters.vue [hunk 5]
```diff
@@ -13,6 +13,16 @@
     >
       <section role="search">
         <client-only>
+          <b-row
+            v-if="advancedSearchEnabled"
+          >
+            <b-button
+              @click="toggleAdvancedSearch"
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/components/search/SideFilters.vue]: Done. Additionally noted that both SearchQueryBuilder instances and the SearchQueryBuilderRule form fields (when multiple rules) had duplicate id's. Those now all have identical id's

File: packages/portal/src/components/search/SideFilters.vue [hunk 6]
```diff
@@ -13,6 +13,16 @@
...
+            >
+              {{ $t('search.advanced.show', {'show': showAdvancedSearch ? 'hide' : 'show' }) }}
+            </b-button>
+          </b-row>
+          <slot />
```
[the remark this answers is not in the document]
Comment (LeoniePeters) [reply packages/portal/src/components/search/SideFilters.vue:16]: No, toggle is now rendered via the slot.

**Review by rwd — APPROVED** [review]

**Pull request merged**